lib/hexapdf/type/image.rb

```diff
@@ -254,6 +257,48 @@ def png_chunk(type, data = '')
         [data.length].pack("N") << type << data << [Zlib.crc32(data, Zlib.crc32(type))].pack("N")
       end
 
+      # Writes the image as TIFF to the given IO stream.
+      def write_tiff(io, info)
+        stream = stream_source
+
+        # header
+        io << ['49492A00'].pack('H*')
+
+        # offset to IFD
+        pad_size = stream.length % 2
+        io << [stream.length + 8 + pad_size].pack('L')
+
+        # data
+        io << Filter.string_from_source(stream)
```

Comment:
It's better to use something like this:

~~~ruby
while stream.alive? && (chunk = stream.resume)
  io << chunk
end
~~~

This will not create a potentially big string but reads and writes the file in chunks.